test-graphql/product-helpers-test.js

```diff
@@ -0,0 +1,38 @@
+import assert from 'assert';
+import Config from '../src-graphql/config';
+import Client from '../src-graphql/client';
+import ProductHelpers from '../src-graphql/product-helpers';
+import singleProductFixture from '../fixtures/product-fixture';
+import fetchMock from './isomorphic-fetch-mock'; // eslint-disable-line import/no-unresolved
+
+suite('product-helpers-test', () => {
+  const productHelpers = new ProductHelpers();
+  const config = new Config({
+    domain: 'single-product.myshopify.com',
+    storefrontAccessToken: 'abc123'
+  });
+
+  const client = new Client(config);
+
+  test('it returns the variant based on options given', () => {
+    fetchMock.postOnce('https://single-product.myshopify.com/api/graphql', singleProductFixture);
```

Comment:
Yeah I don't like how it's generating the model either. I can create a separate PR in `graphql-js-client` to export `decode` so I can properly generate the models.